File: week7/py/rearrange.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))

    for i in range(0, n, 2):
        print(a[i], end=" ")
    for i in range(1, n, 2):
        print(a[i], end=" ")
    print()

def main():
    tst_cnt = int(input())

    while tst_cnt > 0:
        solve()
        tst_cnt -= 1

if __name__ == "__main__":
    main()

File: week7/cpp/rearrange.cpp
#include <bits/stdc++.h> // Include all standard C++ libraries
using namespace std; // Use the standard namespace

void solve(){
		
	int n;	cin>>n;

	int a[n];	

	for(int i=0; i<n; i++){
		cin>>a[i];
	}

	for(int i=0; i<n; i+=2){
		cout<<a[i]<<" ";
	}
	for(int i=1; i<n; i+=2){
		cout<<a[i]<<" ";
	}
	cout<<"\n";
}

int main(){
	
	int tstCnt; // Declare an integer variable for the number of test cases

	cin>>tstCnt; // Read the number of test cases from the input

	while(tstCnt--){ // Loop for each test case

		solve(); // Call the solve function

	}

	return 0;
}

